bgpd/bgp_route.c

```diff
@@ -12659,6 +12667,35 @@ static int bgp_table_stats_single(struct vty *vty, struct bgp *bgp, afi_t afi,
 		if (!json)
 			vty_out(vty, "\n");
 	}
+
+	switch (afi) {
+	case AFI_IP:
+		bitlen = IPV4_MAX_BITLEN;
+		break;
+	case AFI_IP6:
+		bitlen = IPV6_MAX_BITLEN;
+		break;
+	case AFI_L2VPN:
+		bitlen = EVPN_ROUTE_PREFIXLEN;
+		break;
+	default:
+		break;
+	}
+
+	if (json) {
+		json_bitlen = json_object_new_array();
+
+		for (i = 0; i <= bitlen; i++) {
+			struct json_object *ind_bit = json_object_new_object();
+
+			snprintf(temp_buf, sizeof(temp_buf), "%u", i);
+			json_object_int_add(ind_bit, temp_buf,
```

Comment:
nit: Maybe it would be worth creating JSON per _mask_ only if non-zero? Otherwise, it's good to have ALL in place if not looking by eyes (automation).

Reply:
+1

Reply:
I fundamentally disagree.  People looking for stuff in the json output would need to test for existence before they look at the values.

Reply:
:heavy_check_mark:

Reply:
Donald, 

This is what I meant:

we are declaring the array for max length+1 and iterating over it, which in this case is 129. Now instead of printing all the values that are present in the array (129), print/add to json only if the count for a particular prefix length is non zero. Otherwise, we will have a large number of non-zeros output in the json output.